manim/animation/creation.py

```diff
@@ -668,3 +672,202 @@ def __init__(
             )
         )
         super().__init__(*anims, **kwargs)
+
+
+class AddTextLetterByLetterWithCursor(AddTextLetterByLetter):
+    """Similar to :class:`~.AddTextLetterByLetter` , but with an additional cursor mobject at the end.
+
+    Parameters
+    ----------
+    time_per_char
+        Frequency of appearance of the letters.
+    cursor
+        :class:`~.Mobject` shown after the last added letter.
+    buff
+        Controls how far away the cursor is to the right of the last added letter.
+    keep_cursor_y
+        If ``True``, the cursor's y-coordinate is set to the center of the ``Text`` and remains the same throughout the animation. Otherwise, it is set to the center of the last added letter.
+    leave_cursor_on
+        Whether to show the cursor after the animation.
+    kwargs
+        Additional arguments to be passed to the :class:`~.AddTextLetterByLetter` constructor.
+
+    .. tip::
+        This is currently only possible for class:`~.Text` and not for class:`~.MathTex`.
+
+
+    Examples
+    --------
+
+    .. manim:: InsertingTextExample
+        :ref_classes: Blink
+
+        class InsertingTextExample(Scene):
+            def construct(self):
+                text = Text("Inserting", color=PURPLE).scale(1.5).to_edge(LEFT)
+                cursor = Rectangle(
+                    color = GREY_A,
+                    fill_color = GREY_A,
+                    fill_opacity = 1.0,
+                    height = 1.1,
+                    width = 0.5,
+                ).move_to(text[0]) # Position the cursor
+
+                self.play(AddTextLetterByLetterWithCursor(text, cursor))
+                self.play(Blink(cursor, how_many_times=2))
+
+    """
+
+    def __init__(
+        self,
+        text: Text,
+        cursor: Mobject,
+        buff: float = 0.1,
+        keep_cursor_y: bool = True,
+        leave_cursor_on: bool = True,
+        suspend_mobject_updating: bool = False,
```

Comment:
Anything that is simply passed down to the super call (such as `rate_func`, `run_time`, etc.) should not be written in the constructor as well: if the user wants to pass these arguments, they should be passed as `kwargs`.
This also applies to the other classes.